sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/TungstenAggregationIterator.scala

```diff
@@ -170,10 +232,27 @@ class TungstenAggregationIterator(
     val bufferRowSize: Int = bufferSchema.length
 
     val genericMutableBuffer = new GenericMutableRow(bufferRowSize)
-    val unsafeProjection =
-      UnsafeProjection.create(bufferSchema.map(_.dataType))
-    val buffer = unsafeProjection.apply(genericMutableBuffer)
-    initialProjection.target(buffer)(EmptyRow)
+    // TODO(josh): figure out whether we have to use
+    val useUnsafeBuffer = bufferSchema.map(_.dataType).forall(UnsafeRow.isMutable)
+
+    val buffer =  /* if (useUnsafeBuffer) */ {
```

Comment:
When we do hash-based aggregation, the buffer is an UnsafeRow anyway. So, I guess it is fine to let InterpretedAggregate functions to work with UnsafeRows directly. When a developer adds an new InterpretedAggregate function whose aggregate buffer data types are compatible with TungstenAggregate, if he/she uses `update`, he/she will get the exception while testing the function, right? If so, it is safe to say that the code committed in master will not use `update`.